src/Ads/AccountService.php

```diff
@@ -215,46 +216,52 @@ public function get_billing_status(): array {
 	/**
 	 * Gets the ads account access status.
 	 *
-	 * @return bool
+	 * @return array
 	 * @throws Exception When email is not present or invite link is empty when email present and has no account access.
 	 */
 	public function get_ads_account_has_access() {
-		try {
-			// Check if ads id is present.
-			$ads_id = $this->options->get_ads_id();
-
-			if ( empty( $ads_id ) ) {
-				return [
-					'has_access'  => false,
-					'invite_link' => '',
-				];
-			}
+		// Check if ads id is present.
+		$ads_id = $this->options->get_ads_id();
 
-			$connection_status  = $this->container->get( Connection::class )->get_status();
-			$email              = $connection_status['email'] ?? '';
-			$accept_invite_link = $this->options->get( OptionsInterface::ADS_BILLING_URL, '' );
+		if ( empty( $ads_id ) ) {
+			throw new Exception( __( 'Ads id not present', 'google-listings-and-ads' ) );
+		}
 
-			// If no email, means google account is not connected.
-			if ( empty( $email ) ) {
-				throw new Exception( 'Google account is not connected' );
-			}
+		$connection_status  = $this->container->get( Connection::class )->get_status();
+		$email              = $connection_status['email'] ?? '';
+		$accept_invite_link = $this->options->get( OptionsInterface::ADS_BILLING_URL, '' );
```

Comment:
We don't need to check this option until after confirming that we have an email address.